kernel/posix/semaphore.c

```diff
@@ -51,6 +53,9 @@ int sem_init(sem_t *semaphore, int pshared, unsigned int value)
 		return -1;
 	}
 
+	/*
+	 * XXX: this is not in the standard
+	 */
 	if (pshared != 0) {
```

Comment:
This can be changed to just an assertion -- setting `ENOSYS` here is indeed wrong.

Reply:
I agree with you there too, just wanted to point it out with a comment to get some feedback on it.